Rid ourselves of `pipenv`, requirements files, dev containers and other unecessary complexity (#914)
datalab-org/datalab: INSTALL.md

#### Python setup

The next step is to set up a Python environment that contains all of the required dependencies with the correct versions.
You will need Python 3.10 or higher to run *datalab*; we recommend using a tool
to manage Python versions on your machine, to avoid breakages based on your OS's Python versioning (e.g., [`pyenv`](https://github.com/pyenv/pyenv) or [`uv`](https://github.com/astral-sh/uv)).

##### Installation with `uv` or `venv`

We recommend using [`uv`](https://github.com/astral-sh/uv) (see the linked repository or https://docs.astral.sh/uv for installation instructions) for managing your *datalab* installation.

You could also use the standard library `venv` module, but this will not allow you to install pinned dependencies as easily, and is significantly slower than `uv`.

1. Create a virtual environment for *datalab*, ideally inside the `pydatalab` directory.

>     - For `uv`, this will be done automatically on installation.

Reviewer: ```suggestion
    - When installing using `uv sync`, this will be done automatically on installation.
```

Because `uv pip install...` doesn't create the venv

    - For `venv`, this can be done with `python -m venv .venv`.
    - Either way, you will be left with a folder called `.venv` in your `pydatalab` directory that bundles an entire Python environment.
2. Activate the virtual environment (again, optional for `uv`) and install dependencies. One can either use the loosely pinned dependencies in `pyproject.toml`, or the locked versions in `uv.lock`.

=== "Installation with `uv`"

    ```shell
    # EITHER: Install all dependencies with locked versions
    uv sync --all-extras --dev
    # OR: Install all dependencies with loosely pinned versions

>     uv pip install -e '.[all,dev]'

Reviewer: This line gives a warning because `dev` doesn't exist now, and I guess the dev requirements from don't get installed when using this option.

>     uv pip install -e '.[all,dev]'

Author: ```suggestion
    uv pip install -e '.[all]'
```

    ```shell
    source .venv/bin/activate
    # Install all dependencies with loosely pinned versions
    pip install -e '.[all, dev]'
    ```

#### Running the development server

1. Run the server from the `pydatalab` folder with either:

=== "Launching with `uv`"

    ```shell
    cd pydatalab
    uv run flask --app 'pydatalab:main' run --reload
    ```

=== "Launching with `venv`"

    ```shell

>     source .venv/bin/activate

Author: ```suggestion
    cd pydatalab
    source .venv/bin/activate
```

    ```shell
    source .venv/bin/activate
    # Install all dependencies with loosely pinned versions

>     pip install -e '.[all, dev]'

Author: ```suggestion
    pip install -e '.[all]'
```

    ```

#### Running the development server

1. Run the server from the `pydatalab` folder with either:

=== "Launching with `uv`"

    ```shell
    cd pydatalab

>     uv run flask --app 'pydatalab:main' run --reload

Reviewer: ```suggestion
    uv run flask --app 'pydatalab.main' run --reload -p 5001
```

    ```

=== "Launching with `venv`"

    ```shell
    cd pydatalab
    source .venv/bin/activate

>     flask --app 'pydatalab:main' run --reload

Author: ```suggestion
    flask --app 'pydatalab.main' run --reload --port 5001
```
